postprocess.ts

```diff
@@ -4,14 +4,57 @@ import {
   writeCSV,
 } from "https://deno.land/x/flat@0.0.15/mod.ts";
 
+// Check if a filename argument is provided
+if (Deno.args.length < 1) {
+  console.error("Error: Please provide a filename as an argument.");
+  Deno.exit(1);
+}
+
 const filename = Deno.args[0];
-const json = await readJSON(filename);
+
+// Read JSON and handle potential errors
+let json;
+try {
+  json = await readJSON(filename);
+} catch (error) {
+  console.error(`Error reading JSON file: ${error.message}`);
+  Deno.exit(1);
+}
+
+// Validate JSON structure
+if (!json.main || !json.main.temp || !json.weather || !json.weather[0] || !json.weather[0].main) {
+  console.error("Error: JSON does not contain the expected structure.");
+  Deno.exit(1);
+}
 
 const newFilename = "history.csv";
-const history = await readCSV(newFilename);
+let history = [];
 
-await writeCSV(newFilename, [{
+// Read historical data and handle potential errors
+try {
+  history = await readCSV(newFilename);
+} catch (error) {
+  if (error instanceof Deno.errors.NotFound) {
+    console.warn("Warning: history.csv not found. Creating a new file.");
+  } else {
+    console.error(`Error reading history.csv: ${error.message}`);
+    Deno.exit(1);
+  }
```

Comment:
this error will be shown without any error handling as well (and deno will exit with non-zero code)